ext/debug/segger/systemview/SEGGER_SYSVIEW_Conf.h

```diff
@@ -1,11 +1,19 @@
 #pragma once
 
 #include <stdint.h>
+#include <irq.h>
 
 #define SEGGER_SYSVIEW_GET_TIMESTAMP sysview_get_timestamp
 #define SEGGER_SYSVIEW_GET_INTERRUPT_ID sysview_get_interrupt
 
 uint32_t sysview_get_timestamp(void);
 uint32_t sysview_get_interrupt(void);
+extern uint32_t sysview_irq_key;
 
-#define SEGGER_SYSVIEW_RTT_BUFFER_SIZE 4096
+#define SEGGER_SYSVIEW_RTT_BUFFER_SIZE CONFIG_SEGGER_SYSVIEW_RTT_BUFFER_SIZE
+
+// Lock SystemView (nestable)
+#define SEGGER_SYSVIEW_LOCK()             sysview_irq_key = irq_lock();
```

Comment:
Where is `SEGGER_SYSVIEW_LOCK` and `SEGGER_SYSVIEW_UNLOCK` called from? I can't see that from the diff.

Reply:
These are locking functions used by SystemView when it logs data for profiling.

Reply:
If these macros are supposed to be nestable, how would `sysview_irq_key` be preserved from the second call to `SEGGER_SYSVIEW_LOCK()` onwards?

Reply:
ok, I fixed it